scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala

```diff
@@ -289,17 +285,17 @@ abstract class FileSource extends SchemedSource with LocalSourceOverride with Hf
   }
 
   protected def createHdfsReadTap(hdfsMode: Hdfs): Tap[JobConf, _, _] = {
-    val taps: List[Tap[JobConf, RecordReader[_, _], OutputCollector[_, _]]] =
+    val taps =
       goodHdfsPaths(hdfsMode)
-        .toList.map { path => CastHfsTap(createHfsTap(hdfsScheme, path, sinkMode)) }
+        .iterator
+        // some paths deemed "good" may actually be empty, and hadoop's FileInputFormat
+        // doesn't like that. So we filter them away here.
+        .filter { p => FileSource.globHasNonHiddenPaths(p, hdfsMode.conf) }
+        .map { path => CastHfsTap(createHfsTap(hdfsScheme, path, sinkMode)) }
+        .toList
+
     taps.size match {
```

Comment:
can we change this to:
```scala
taps match {
  case  Nil => new IterableSource[Any](Nil).createTap(Read)(hdfsMode).asInstanceOf[Tap[JobConf, _, _]]
  case one :: Nil => one
  case many => new ScaldingMultiSourceTap(many)
}
```
rather than using size?